runtime/lua/lsp/callbacks.lua

```diff
@@ -0,0 +1,208 @@
+-- Implements the following default callbacks:
+--  textDocument/publishDiagnostics
+--  notification: textDocument/didOpen
+--  notification: textDocument/willSave
+--  TODO: textDocument/willSaveWaitUntil
+--  notification: textDocument/didSave
+--  notification: textDocument/didClose
+--  IN PROGRESS: textDocument/completion
+--  textDocument/hover
+--  TODO: textDocument/signatureHelp
+--  textDocument/references
+--  textDocument/documentHighlight
+--  TODO: textDocument/documentSymbol
+--  TODO: textDocument/formatting
+--  TODO: textDocument/rangeFormatting
+--  TODO: textDocument/onTypeFormatting
+--  textDocument/definition
+--  TODO: textDocument/codeAction
+--  TODO: textDocument/codeLens
+--  TODO: textDocument/documentLink
+--  TODO: textDocument/rename
+--
+--  TODO: completionItem/resolve
+--
+--  TODO: codeLens/resolve
+--
+--  TODO: documentLink/resolve
+
+local log = require('runtime.lua.log')
+local util = require('runtime.lua.builtin_util')
+local lsp_util = require('runtime.lua.lsp.util')
```

Comment:
I only point-sampled a few, but ISTM like those functions are:

1. Very short.
2. Only used in this file.

Because of the generic name (`util`), we also have to prefix it with `lsp_util`, even though they have nothing to do with LSP AFAIK.

How about just inlining them as locals?